### ISSUE TITLE
Build failure with v 4.0.0 (error: image must not be null)

### ISSUE THREAD

**Issue by miduch** [usefulness/easylauncher-gradle-plugin#204]
Hi, I'm getting following error with plugin when using in a hello world (jetpack compose) project with minSdk 24 & targetSdk 30

Execution failed for task ':app:easylauncherDebug'.
> image must not be null

in app/build.gradle i've 

plugins {
    id 'com.starter.easylauncher' version '4.0.0'
}

Example Project setup/ configs

    defaultConfig {
        applicationId "com.example.ribbontest"
        minSdk 24
        targetSdk 30
        versionCode 1
        versionName "1.0"

        vectorDrawables {
            useSupportLibrary true
        }
    }
 

<img width="297" alt="Screenshot 2021-08-14 at 22 40 05" src="https://user-images.githubusercontent.com/22243296/129458401-9618801e-e486-4b83-b5e2-0ebbccde9173.png">

Also tried to specify iconNames like below but it didn't help  
easylauncher {
        iconNames = ["@mipmap/ic_launcher", "@mipmap/ic_launcher_round"]
    }

Does work ONLY if full folder name is specified
easylauncher {
        iconNames = ["@mipmap-anydpi-v26/ic_launcher_round"]
    }

But it causes then another issue/ app crashes if i try to use that icon in app somewhere e.g. in splash screen  

    java.lang.IllegalArgumentException: Only VectorDrawables and rasterized asset types are supported ex. PNG, JPG
        at androidx.compose.ui.res.PainterResources_androidKt.loadVectorResource(PainterResources.android.kt:85)

```
Image(
            modifier = Modifier.scale(scale.value),
            painter = painterResource(id = R.mipmap.ic_launcher), // same result with R.mipmap.ic_launcher_round, R.drawable.ic_launcher_foreground
            contentDescription = null
        )

```

**Comment by miduch** [comment]
Project is just a hello world (jetpack compose) project with minSdk 24 & targetSdk 30

**Comment by mateuszkwiecinski** [comment]
Hey @miduch 👋 
Thanks for reporting the issue. 
I'm guessing the plugin doesn't support `webp`s and I'd have to do some research if adding support for such would be feasible 👀 
As a workaround I can only suggest using `png`s as launcher images.

Regarding the `IllegalArgumentException`: would you be able to share full repro project? I'm curious what's the format of produced artifacts 👀

**Comment by miduch** [comment]
> Hey @miduch 👋
> Thanks for reporting the issue.
> I'm guessing the plugin doesn't support `webp`s and I'd have to do some research if adding support for such would be feasible 👀
> As a workaround I can only suggest using `png`s as launcher images.
> 
> Regarding the `IllegalArgumentException`: would you be able to share full repro project? I'm curious what's the format of produced artifacts 👀

zip with all build files 
https://github.com/miduch/RibbonTest/blob/main/RibbonTest_1.zip 

example project 
https://github.com/miduch/RibbonTest

**Comment by miduch** [comment]
> I'm guessing the plugin doesn't support `webp`s and I'd have to do some research if adding support for such would be feasible 👀
> As a workaround I can only suggest using `png`s as launcher images.
> 

android studio artic fox by default is creating webp for new project when created from wizard. if plugin doesn't support webp (yet?) then atleast this should be mentioned in plugin's ReadMe or somewhere

**Comment by mateuszkwiecinski** [comment]
Hey @miduch Thank you very much for sample repository 🙏 
The webp support should be fixed in #207 and I'll trigger the release pipeline soon. 

Regarding the error
>java.lang.IllegalArgumentException: Only VectorDrawables and rasterized asset types are supported ex. PNG, JPG
    at androidx.compose.ui.res.PainterResources_androidKt.loadVectorResource(PainterResources.android.kt:85)

it is not related to this plugin, but rather to the fact you're trying to load an adaptive icon resource as an image which is not supported. I don't know if that's Composes thing or the Android Framework as a whole, but that's a part I can't help with. If you run your app on a device with api below 26 it would work because it then loads webp resource variant of `R.mipmap.ic_launcher`

**Comment by miduch** [comment]
java.lang.IllegalArgumentException

Compose can load same image IF I don't apply this plugin, exception starts to come only after plugin is applied.

**Comment by miduch** [comment]
> java.lang.IllegalArgumentException
> 
> Compose can load same image IF I don't apply this plugin, exception starts to come only after plugin is applied. 

But let's wait and see how does it work with this new upcoming release of plugin. 

Thanks!

**Comment by mateuszkwiecinski** [comment]
>Compose can load same image IF I don't apply this plugin, exception starts to come only after plugin is applied.

It's been crashing for me with exactly the same stacktrace 👀 
The only difference I see after applying the plugin is a ribbon visible on the launcher:
![image](https://user-images.githubusercontent.com/36954793/130314591-44913cf8-e0a6-4f88-9186-c7473091b041.png)

**Comment by miduch** [comment]
> > Compose can load same image IF I don't apply this plugin, exception starts to come only after plugin is applied.
> 
> It's been crashing for me with exactly the same stacktrace 👀
> The only difference I see after applying the plugin is a ribbon visible on the launcher:
> ![image](https://user-images.githubusercontent.com/36954793/130314591-44913cf8-e0a6-4f88-9186-c7473091b041.png)
> 
> FYI: Version `4.1.0` should be already available for download


when applying plugin, following is the file which seems to be used with 

`painter = painterResource(id = R.drawable.ic_launcher_foreground)
`
-> app/build/generated/easylauncher/res/debug/drawable-anydpi-v26/ic_launcher_foreground.xml

Compose doesn't seem to like <layer-list , wants either <vector or png e.t.c. 

```
<?xml version="1.0" encoding="utf-8"?>
<layer-list xmlns:android="http://schemas.android.com/apk/res/android">

    <item android:drawable="@drawable/easy_ic_launcher_foreground" />

    <item
        android:width="72dp"
        android:height="72dp"
        android:drawable="@drawable/colorribbonfilter_0_ic_launcher_foreground"
        android:gravity="center"
        />

</layer-list>

```

**Comment by mateuszkwiecinski** [comment]
>painter = painterResource(id = R.drawable.ic_launcher_foreground)

Ah, I understood you aimed to load `R.mipmap.ic_launcher` not `R.drawable.ic_launcher_foreground`

`ic_launcher` on >26 api levels is an adaptive icon resource, which I do't 
`ic_launcher_foreground` and `ic_launcher_round` is a `layer-list` drawable, which doesn't seem to be supported by Compose

I don't know what you aim for, but the `layer-list` drawable is the way the plugin injects itself to adds overlay to app icon. If you loaded the `ic_launcher_foreground` resource to plain old `ImageView` - it would work as expected, so it seems like the issue is with Compose. Feel free to report an issue to Compose Team and see what they reply :)
